sdk/contracts/ShieldedPoolWithBlocklist.sol

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import {CompliantVerifier} from "./generated/CompliantVerifier.sol";
+import {MerkleTreeWithHistory} from "./MerkleTreeWithHistory.sol";
+import {MerkleTreeForBlocklist} from "./MerkleTreeForBlocklist.sol";
+import {SwapExecutor} from "./SwapExecutor.sol";
+
+contract ShieldedPoolWithBlocklist is MerkleTreeWithHistory {
```

Comment:
There is so much duplication in this contract with ShieldedPool.sol I wonder if down the track we can eventually extract out similarities? Maybe pull the different structs from a lib contract and inject the bits that change.

Reply:
yeah, have similar thoughts, I'll refactor it